cloud/amazon/ec2_vol.py

```diff
@@ -325,8 +392,40 @@ def attach_volume(module, ec2, volume, instance):
         except boto.exception.BotoServerError, e:
             module.fail_json(msg = "%s: %s" % (e.error_code, e.error_message))
 
+        modify_dot_attribute(module, ec2, instance, device_name)
+
     return volume, changed
 
+
+def modify_dot_attribute(module, ec2, instance, device_name):
+
+    delete_on_termination = module.params.get('delete_on_termination')
+    changed = False
+
+    try:
+        instance.update()
+        dot = instance.block_device_mapping[device_name].delete_on_termination
+    except boto.exception.BotoServerError, e:
+        module.fail_json(msg = "%s: %s" % (e.error_code, e.error_message))
+  
+    if delete_on_termination ^ dot:
```

Comment:
It updates instance object and reads the current value of delete_on_termination attribute.
Added doc string to function for more clarity.
What do you mean by more clarity? Missing comments or "dot" usage in function name?